File: app/src/main/java/apextechies/starbasket/activity/OrderDetailsActivity.java
package apextechies.starbasket.activity;

import android.os.Bundle;
import android.preference.Preference;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;


import org.json.JSONException;
import org.json.JSONObject;

import apextechies.starbasket.R;
import apextechies.starbasket.adapter.OrderDetailsAdapter;
import apextechies.starbasket.model.OrderDetailsModel;
import apextechies.starbasket.model.OrderModel;
import apextechies.starbasket.retrofit.ApiUrl;

public class OrderDetailsActivity extends BaseActivity  {
	private static final int RC_DETAILS = 1;
	private OrderDetailsAdapter mAdapter;
	private OrderModel mOrder;
	
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		setContentView(R.layout.activity_order_details);
		super.onCreate(savedInstanceState);
		
		mAdapter = new OrderDetailsAdapter();
		RecyclerView orderRV = (RecyclerView) findViewById(R.id.rv_order);
		orderRV.setAdapter(mAdapter);
		
		//mOrder = (OrderModel) getIntent().getSerializableExtra(EXTRA_DATA);
		
		/*ApiTask.builder(this)
			.setGET()
			.setUrl(ApiUrl.ORDER_DETAILS + mOrder.getOrderId() + "/" + new Preference(this).getSessionKey())
			.setProgressMessage(R.string.loading_order_details)
			.setRequestCode(RC_DETAILS)
			.setResponseListener(this)
			.exec();*/
	}
	
	/*@Override
	public void onSuccess(JSONObject response, int requestCode, Bundle savedData) throws JSONException {
		JSONObject data = response.getJSONObject("data").getJSONArray("orders").getJSONObject(0);
		OrderDetailsModel order = new OrderDetailsModel(data);
		
		((TextView) findViewById(R.id.tv_order_id)).setText(getString(R.string.format_order_id, mOrder.getOrderId()));
		((TextView) findViewById(R.id.tv_total)).setText(getString(R.string.format_price, mOrder.getTotal()));
		((TextView) findViewById(R.id.tv_order_status)).setText(mOrder.getOrderStatus());
		if (order.getPromo() > 0) {
			((TextView) findViewById(R.id.tv_promo)).setText(getString(R.string.format_price, order.getPromo()));
		} else {
			findViewById(R.id.ll_promo).setVisibility(View.GONE);
		}
		
		mAdapter.addItems(order.getItemList());
	}
	
	@Override
	public void onFailure(int requestCode, Bundle savedData) {
		
	}*/
}

File: app/src/main/java/apextechies/starbasket/activity/OrderDetailsActivity.kt
package apextechies.starbasket.activity

import android.content.DialogInterface
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity


import apextechies.starbasket.R
import apextechies.starbasket.adapter.OrderDetailsAdapter
import apextechies.starbasket.common.Utilz
import apextechies.starbasket.model.CommonModel
import apextechies.starbasket.model.UserOrderDataListModel
import apextechies.starbasket.retrofit.DownlodableCallback
import apextechies.starbasket.retrofit.RetrofitDataProvider
import kotlinx.android.synthetic.main.activity_order_details.*
import kotlinx.android.synthetic.main.toolbar.*

class OrderDetailsActivity : AppCompatActivity() {
    private var mAdapter: OrderDetailsAdapter? = null
    var retrofitDataProvider: RetrofitDataProvider?= null


    override fun onCreate(savedInstanceState: Bundle?) {
        setContentView(R.layout.activity_order_details)
        super.onCreate(savedInstanceState)

        setSupportActionBar(toolbarr)
        supportActionBar!!.setDisplayShowHomeEnabled(true)
        supportActionBar!!.title = "Order Details"

        retrofitDataProvider = RetrofitDataProvider((this))

        mAdapter = OrderDetailsAdapter()
        rv_order.adapter = mAdapter

       var  mOrder: UserOrderDataListModel? = intent.getParcelableExtra("extra_data")

        tv_order_id.text = (getString(R.string.format_order_id, mOrder!!.transaction_id));
        tv_total.text = "₹ "+mOrder.total_price
        tv_order_status.text = Utilz.getStatus(mOrder.order_status);


        mAdapter!!.addItems(mOrder.varientdetails!!)

        toolbarr.setNavigationOnClickListener {
            finish()
        }

        tv_cancel.setOnClickListener {

            showDilog(mOrder!!.transaction_id!!.replace("#", ""))
        }

        date.text = mOrder.order_date
    }
    private fun showDilog(item: String) {
        val builder = AlertDialog.Builder(this)

        if (title != null) builder.setTitle(title)

        builder.setMessage("Are you sure you want to delete this item from your cart?")
        builder.setPositiveButton("OK", object : DialogInterface.OnClickListener {
            override fun onClick(dialog: DialogInterface?, which: Int) {
                cancelOrder(item)
            }

        })
        builder.setNegativeButton("Cancel", null)
        builder.show()
    }
    private fun cancelOrder(transaction_id: String?) {

        retrofitDataProvider!!.cancelOrder(transaction_id, object : DownlodableCallback<CommonModel> {
            override fun onSuccess(result: CommonModel?) {
                if (result!!.status.equals("true")){
                    startActivity(Intent(this@OrderDetailsActivity, OrderActivity::class.java))
                    finish()
                }
            }

            override fun onFailure(error: String?) { }

            override fun onUnauthorized(errorNumber: Int) { }
        })
    }


}
